rs/memory_tracker/src/lib.rs

```diff
@@ -31,24 +32,27 @@ fn new_signal_handler_available() -> bool {
 // size must be a multiple of PAGE_SIZE.
 #[derive(Clone)]
 pub struct MemoryArea {
-    // base address of the tracked memory area
+    // Base address of the tracked memory area.
     addr: usize,
-    // size of the tracked memory area
-    size: Cell<usize>,
```

Comment:
Are you sure about removing the `Cell`? It seems to me that it was used to allow changing the size of the memory area but without being able to mutate the base `addr`. Now, during the `expand` for example one could change the `addr` which we might not want to. I'm not sure if this property (not being able to change the base `addr` of a `MemoryArea`) is super critical though.